src/main/kotlin/com/team4099/robot2020/subsystems/Shooter.kt

```diff
@@ -0,0 +1,83 @@
+import com.revrobotics.ControlType
+import com.team4099.lib.motorcontroller.SparkMaxControllerFactory
+import com.team4099.lib.subsystem.Subsystem
+import com.team4099.robot2020.config.Constants
+import java.lang.Math.abs
+
+object Shooter : Subsystem {
+
+    private val masterSparkMax = SparkMaxControllerFactory.createDefaultSparkMax(
+            Constants.Shooter.MASTER_SPARKMAX_ID)
+    private val slaveSparkMax = SparkMaxControllerFactory.createPermanentSlaveSparkMax(
+            Constants.Shooter.SLAVE_SPARKMAX_ID, masterSparkMax)
+
+    // set these later
+    private var targetSpeed = 0.0
+    private var currentSpeed = 0.0
+    private var speedThreshold = 0.0
+
+    enum class State {
+        SHOOTING, IDLE, ACCELERATING
+    }
+
+    private var currentState = State.IDLE
+
+    init {
+        masterSparkMax.pidController.setP(Constants.Shooter.P_VALUE)
+        masterSparkMax.pidController.setI(Constants.Shooter.I_VALUE)
+        masterSparkMax.pidController.setD(Constants.Shooter.D_VALUE)
+        masterSparkMax.pidController.setFF(Constants.Shooter.F_VALUE)
+
+        // change later
+        masterSparkMax.setSmartCurrentLimit(0)
+
+        slaveSparkMax.inverted = false
+        // slaveSparkMax.setInverted(false)
+    }
+
+    fun setOpenLoop(speed: Double) {
```

Comment:
speed isn't the greatest parameter name for either of these, would recommend `power` and `velocity`